### ISSUE TITLE
Allow users to modify Esbuild configuration

### ISSUE THREAD

**Issue by dimkk** [goldstack/goldstack#125]
**Current behavior**:
Trying to use module which uses node streams, getting this error:
![image](https://user-images.githubusercontent.com/4674044/156729890-3023ad43-0da2-4098-b5da-6eef4e77702f.png)

After a little investigation - all values are hard coded in @goldstack/template-lambda-api/src/templateLambdaApiBuild.ts
![image](https://user-images.githubusercontent.com/4674044/156729241-d5f8af2a-a74c-4cbe-98c2-ff7ca7633f27.png)
And to avoid error above, I need to pass - "platform":"node" prop to esbuild.

**Proposed behavior:**
Add the ability to extend this config, using additional property in LambdaConfig type, which will merge with default ones.
Something like this:
![image](https://user-images.githubusercontent.com/4674044/156729718-9ce22027-e387-4ab2-a2ce-a489db1e9fbd.png)

I will do a PR soon, so RFC.

**Comment by mxro** [comment]
Thanks for raising this issue. That's a great idea. Indeed, there is a left-over file [`scripts/build.ts`](https://github.com/goldstack/goldstack/blob/master/workspaces/templates/packages/serverless-api/scripts/build.ts) in the template that is not used (originally got there since it is based on the [lambda-express](https://github.com/goldstack/goldstack/tree/master/workspaces/templates/packages/lambda-express) template).

While it is possible to pass through esbuild configuration parameters through the `config` object here (instance of [LambdaConfig](https://github.com/goldstack/goldstack/blob/master/workspaces/templates-lib/packages/utils-aws-lambda/src/generate/collectLambdasFromFiles.ts#L9)), I think that may not be the preferrable approach here, since that is generated for every lambda individually.

If we don't need to have different esbuild configurations for the different Lambdas, it maybe easier to provide just one esbuild config that is used by all Lambdas?

It looks like esbuild doesn't support a configuration file, see https://github.com/evanw/esbuild/issues/39 but I guess it would be trivial to load one from a JSON? Don't think there would be need for using something like [esbuild-config](https://github.com/bpierre/esbuild-config). Instead, we could do something like:

```javascript
    mkdir('-p', getOutDirForLambda(config));
    const esbuildConfig = read('./esbuild.config.json');
    await build({
      plugins: [pnpPlugin()],
      bundle: true,
      entryPoints: [`${defaultRoutesPath}/${config.relativeFilePath}`],
      external: ['aws-sdk'],
      minify: true,
      format: 'cjs',
      target: 'node12.0',
      sourcemap: true,
      outfile: getOutFileForLambda(config),
      ...esbuildConfig,
    });
```

If per-Lambda config would be required, maybe allow defining esbuild configs alongside the handler files, such as:

```
src/routes/hello.ts
src/routes/hello.esbuild.config.json
```

Would JSON files be sufficient to supply all config options or are some of them JS objects and it would need to be JS source files (such as for Webpack).

**Comment by mxro** [comment]
Oh and before I forget, a few things to keep in mind for local Development as per https://github.com/goldstack/goldstack/blob/master/CONTRIBUTING.md - biggest caveat is that when doing local dev in the goldstack repository, one needs to use the command `yarn template-ts ...` instead of `yarn template` (to allow the script to pick up any modifications to TS sources).

**Comment by dimkk** [comment]
> Would JSON files be sufficient to supply all config options or are some of them JS objects and it would need to be JS source files (such as for Webpack).

I believe json is enough, user can customize build/deploy process and include additional steps for its specific use case.

**Comment by mxro** [comment]
Released with [v0.3.19](https://github.com/goldstack/goldstack/releases/tag/v0.3.19) - for already created projects, [`@goldstack/template-lambda-api`](https://www.npmjs.com/package/@goldstack/template-lambda-api) can be upgraded to version `0.4.32`.

**Comment by mxro** [comment]
Closing this issue for now - please reopen if there is anything more we need to do!
